### PR TITLE
FIPS 140-3 changes for master

### PR DESCRIPTION
**Pull request by paulidale**
Clone of #19486 but for master.  A couple of trivial cherry-pick conflicts to resolve.

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Will approve once @t-j-h  suggested changes are committed.

**Comment by slontis** [comment]
try again :)

**Comment by paulidale** [comment]
Both PRs passing tests now.

File: util/libcrypto.num [hunk 1]
```diff
@@ -5469,6 +5469,9 @@ OSSL_sleep                              ?	3_2_0	EXIST::FUNCTION:
 OSSL_get_thread_support_flags           ?	3_2_0	EXIST::FUNCTION:
 OSSL_set_max_threads                    ?	3_2_0	EXIST::FUNCTION:
 OSSL_get_max_threads                    ?	3_2_0	EXIST::FUNCTION:
+EVP_RAND_CTX_up_ref                     ?	3_0_7	EXIST::FUNCTION:
+RAND_set0_public                        ?	3_0_7	EXIST::FUNCTION:
+RAND_set0_private                       ?	3_0_7	EXIST::FUNCTION:
```
Comment (t8m) [inline util/libcrypto.num:5474]: Wrong version numbers - The correct is 3_1_0

File: doc/man1/openssl-fipsinstall.pod.in [hunk 2]
```diff
@@ -174,6 +175,17 @@ target machine. Once the self tests have run on the target machine the user
...
 
+This is the default.
+
+=item B<-self_test_oninstall>
+
+The converse of B<-self_test_oninstall>.  The two fields related to the
+"test status indicator" and "MAC status indicator" are written to the
+output configuration file.
+
+This option allows OpenSSL 3.0 FIPS providers to be installed so that
```
Comment (slontis) [inline doc/man1/openssl-fipsinstall.pod.in:186]: Maybe we shoud just omit this line?

File: util/libcrypto.num (hunk 1 above)
Comment (paulidale) [reply util/libcrypto.num]: fixed and relocated to a better place.

File: doc/man3/EVP_RAND.pod [hunk 3]
```diff
@@ -30,6 +30,7 @@ EVP_RAND_STATE_ERROR - EVP RAND routines
  void EVP_RAND_free(EVP_RAND *rand);
  EVP_RAND_CTX *EVP_RAND_CTX_new(EVP_RAND *rand, EVP_RAND_CTX *parent);
  void EVP_RAND_CTX_free(EVP_RAND_CTX *ctx);
+ int EVP_RAND_CTX_up_ref(EVP_RAND_CTX *ctx);
```
Comment (slontis) [inline doc/man3/EVP_RAND.pod:33]: Add to history?

Comment (paulidale) [reply doc/man3/EVP_RAND.pod:33]: Added

**Review by slontis — APPROVED** [review]

**Review by t-j-h — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**